### PR TITLE
Add redirects for integration-testing and headers

### PR DESCRIPTION
**Pull request by jieyouxu**
I forgor to add redirects for them from the old URLs to the new URLs.

### CODE REVIEW

File: book.toml [hunk 1]
```diff
@@ -54,3 +54,5 @@ warning-policy = "error"
 "/diagnostics/sessiondiagnostic.html" = "diagnostic-structs.html"
 "/diagnostics/diagnostic-codes.html" = "error-codes.html"
 "/miri.html" = "const-eval/interpret.html"
+"/tests/integration.html" = "tests/ecosystem.html"
+"/tests/headers.html" = "tests/directives.html"
```
Comment (ehuss) [inline book.toml:58]: These need to be relative to the source directory.

```suggestion
"/tests/integration.html" = "ecosystem.html"
"/tests/headers.html" = "directives.html"
```

**Review by ehuss — COMMENTED** [review]
Thanks! ❤️

File: book.toml (hunk 1 above)
Comment (jieyouxu) [reply book.toml:58]: Wait wdym relative to the source directory? Because e.g. directives.html is still under the `tests/` folder no?

Comment (jieyouxu) [reply book.toml]: I changed the right-side to be as suggested, though I can't claim I understand this redirect syntax exactly 🤷

Comment (jieyouxu) [reply book.toml]: Ah okay thanks for the explanation.

**Review by DianQK — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**